### PR TITLE
SYN-6083, SYN-6060: Add unlisted perms

### PR DESCRIPTION
**Pull request by MichaelSquires**
- Updated cortex permdefs to include unlisted permissions

Resolves SYN-6083 and SYN-6060

### CODE REVIEW

**Comment by MichaelSquires** [comment]
I'm not sure all the gates are quite correct. If I wasn't sure (which is most of them), I left it as `cortex`. Please keep an eye on those.

File: synapse/lib/stormtypes.py [hunk 1]
```diff
@@ -7181,6 +7181,19 @@ class LibTrigger(Lib):
                   'returns': {'type': 'str', 'desc': 'The iden of the modified Trigger', }}},
     )
     _storm_lib_path = ('trigger',)
+    _storm_lib_perms = (
+        {'perm': ('trigger', 'add'), 'gate': 'cortex',
+         'desc': 'Controls adding triggers.'},
+        {'perm': ('trigger', 'del'), 'gate': 'cortex',
+         'desc': 'Controls deleting triggers.'},
+        {'perm': ('trigger', 'get'), 'gate': 'cortex',
```
Comment (vEpiphyte) [inline synapse/lib/stormtypes.py:7189]: gated on the view

File: synapse/lib/stormtypes.py [hunk 2]
```diff
@@ -7181,6 +7181,19 @@ class LibTrigger(Lib):
                   'returns': {'type': 'str', 'desc': 'The iden of the modified Trigger', }}},
     )
     _storm_lib_path = ('trigger',)
+    _storm_lib_perms = (
+        {'perm': ('trigger', 'add'), 'gate': 'cortex',
+         'desc': 'Controls adding triggers.'},
```
Comment (vEpiphyte) [inline synapse/lib/stormtypes.py:7186]: gated on the view

Comment (vEpiphyte) [reply synapse/lib/stormtypes.py:7186]: gated on the trigger

File: synapse/lib/stormtypes.py [hunk 3]
```diff
@@ -7181,6 +7181,19 @@ class LibTrigger(Lib):
...
+        {'perm': ('trigger', 'del'), 'gate': 'cortex',
```
Comment (vEpiphyte) [inline synapse/lib/stormtypes.py:7187]: gated on the trigger

File: synapse/lib/stormtypes.py [hunk 4]
```diff
@@ -7181,6 +7181,19 @@ class LibTrigger(Lib):
...
+         'desc': 'Controls deleting triggers.'},
+        {'perm': ('trigger', 'get'), 'gate': 'cortex',
+         'desc': 'Controls listing/retrieving triggers.'},
+        {'perm': ('trigger', 'set'), 'gate': 'cortex',
+         'desc': 'Controls modifying triggers.'},
+        {'perm': ('trigger', 'set', '<name>'), 'gate': 'cortex',
```
[the remark this answers is not in the document]
Comment (vEpiphyte) [reply synapse/lib/stormtypes.py:7193]: `trigger.set` can also be gated on the Trigger iden :|

File: synapse/cortex.py [hunk 5]
```diff
@@ -1542,6 +1567,42 @@ def _initCorePerms(self):
...
+            'desc': 'Controls importing modules requiring root privileges.',
+             'ex': 'storm.asroot.cmd.synapse-misp.privsep'},
+
+            {'perm': ('storm', 'graph', 'add'), 'gate': 'cortex',
+             'desc': 'Controls access to add a storm graph.',
+             'default': True},
+            {'perm': ('storm', 'macro', 'add'), 'gate': 'cortex',
+             'desc': 'Controls access to add a storm macro.',
+             'default': True},
+            {'perm': ('storm', 'macro', 'admin'), 'gate': 'macro',
```
Comment (vEpiphyte) [inline synapse/cortex.py:1589]: gated on the cortex

File: synapse/cortex.py [hunk 6]
```diff
@@ -1542,6 +1567,42 @@ def _initCorePerms(self):
...
+             'desc': 'Controls access to edit/set/delete a storm macro.'},
+            {'perm': ('storm', 'macro', 'edit'), 'gate': 'macro',
```
Comment (vEpiphyte) [inline synapse/cortex.py:1591]: gated on the cortex

File: synapse/lib/stormtypes.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (MichaelSquires) [reply synapse/lib/stormtypes.py:7193]: So what do you think for `trigger.set`? I can leave it as just `view`, use `view|trigger`, or duplicate it (I don't know if that will work though) and have one listed as `view` and the other as `trigger`.

File: synapse/lib/stormtypes.py [hunk 7]
```diff
@@ -8839,6 +8895,16 @@ class LibCron(Lib):
...
     )
     _storm_lib_path = ('cron',)
+    _storm_lib_perms = (
+        {'perm': ('cron', 'add'), 'gate': 'view',
+         'desc': 'Permits a user to create a cron job.'},
+        {'perm': ('cron', 'del'), 'gate': 'cronjob',
+         'desc': 'Permits a user to remove a cron job.'},
+        {'perm': ('cron', 'get'), 'gate': 'cronjob',
+         'desc': 'Permits a user to list cron jobs.'},
+        {'perm': ('cron', 'set'), 'gate': 'cronjob',
```
Comment (vEpiphyte) [inline synapse/lib/stormtypes.py:8905]: `cron.set.creator` is gated at the cortex

File: synapse/lib/stormtypes.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (MichaelSquires) [reply synapse/lib/stormtypes.py]: ok, I think I got this updated properly.

File: synapse/lib/stormtypes.py [hunk 8]
```diff
@@ -7214,6 +7214,24 @@ class LibTrigger(Lib):
...
     _storm_lib_path = ('trigger',)
+    _storm_lib_perms = (
+        {'perm': ('trigger', 'add'), 'gate': 'cortex',
+         'desc': 'Controls adding triggers.'},
+        {'perm': ('trigger', 'del'), 'gate': 'view',
+         'desc': 'Controls deleting triggers.'},
+        {'perm': ('trigger', 'get'), 'gate': 'trigger',
+         'desc': 'Controls listing/retrieving triggers.'},
+        {'perm': ('trigger', 'set'), 'gate': 'view',
+         'desc': 'Controls modifying triggers.'},
```
Comment (vEpiphyte) [inline synapse/lib/stormtypes.py:7225]: This is used for enabling, disabling, and modifying the query of a trigger.

**Review by vEpiphyte — APPROVED** [review]

**Pull request merged**